exercises/strain/canonical-data.json

```
@@ -0,0 +1,155 @@
+          "description": "keeps everything",
+          "property": "keep",
+          "input": {
+            "list": [1, 3, 5],
+            "predicate": "fn(x) -> x % 2 == 1"
+          },
+          "expected": [1, 3, 5]
+        },
+        {
+          "uuid": "950b8e8e-f628-42a8-85e2-9b30f09cde38",
+          "description": "keeps nothing",
+          "property": "keep",
+          "input": {
+            "list": [1, 3, 5],
+            "predicate": "fn(x) -> x % 2 == 0"
+          },
+          "expected": []
+        },
+        {
+          "uuid": "92694259-6e76-470c-af87-156bdf75018a",
+          "description": "keeps first and last",
+          "property": "keep",
+          "input": {
+            "list": [1, 2, 3],
+            "predicate": "fn(x) -> x % 2 == 1"
+          },
+          "expected": [1, 3]
+        },
+        {
+          "uuid": "938f7867-bfc7-449e-a21b-7b00cbb56994",
+          "description": "keeps neither first nor last",
+          "property": "keep",
+          "input": {
+            "list": [1, 2, 3],
+            "predicate": "fn(x) -> x % 2 == 0"
+          },
+          "expected": [2]
+        },
+        {
+          "uuid": "8908e351-4437-4d2b-a0f7-770811e48816",
+          "description": "keeps strings",
+          "property": "keep",
+          "input": {
+            "list": ["apple", "zebra", "banana", "zombies", "cherimoya", "zealot"],
+            "predicate": "fn(x) -> starts_with(x, 'z')"
+          },
+          "expected": ["zebra", "zombies", "zealot"]
+        },
+        {
+          "uuid": "2728036b-102a-4f1e-a3ef-eac6160d876a",
+          "description": "keeps lists",
+          "property": "keep",
+          "input": {
+            "list": [[1, 2, 3], [5, 5, 5], [5, 1, 2], [2, 1, 2], [1, 5, 2], [2, 2, 1], [1, 2, 5]],
+            "predicate": "fn(x) -> contains(x, 5)"
+          },
+          "expected": [[5, 5, 5], [5, 1, 2], [1, 5, 2], [1, 2, 5]]
+        }
+      ]
+    },
+    {
+      "description": "filter list returning only values that do not satisfy the predicate function",
+      "cases": [
+        {
+          "uuid": "ef16beb9-8d84-451a-996a-14e80607fce6",
+          "description": "discard on empty list returns empty list",
+          "property": "discard",
+          "input": {
+            "list": [],
+            "predicate": "fn(x) -> true"
+          },
+          "expected": []
+        },
+        {
+          "uuid": "2f42f9bc-8e06-4afe-a222-051b5d8cd12a",
+          "description": "discards everything",
+          "property": "discard",
+          "input": {
+            "list": [1, 3, 5],
+            "predicate": "fn(x) -> x % 2 == 1"
+          },
+          "expected": []
+        },
+        {
+          "uuid": "ca990fdd-08c2-4f95-aa50-e0f5e1d6802b",
+          "description": "discards nothing",
+          "property": "discard",
+          "input": {
+            "list": [1, 3, 5],
+            "predicate": "fn(x) -> x % 2 == 0"
```

Similar suggestion as with the corresponding keep test case:

```suggestion
            "predicate": "fn(x) -> false"
```